public/app/plugins/datasource/loki/datasource.test.ts

```diff
@@ -357,6 +357,43 @@ describe('LokiDatasource', () => {
       expect(res[1].tags).toEqual(['value2']);
     });
   });
+
+  describe('metricFindQuery', () => {
+    let ds: LokiDatasource;
+    beforeEach(() => {
+      ds = new LokiDatasource(instanceSettings, backendSrv, templateSrvMock);
+      backendSrvMock.datasourceRequest = jest.fn().mockReturnValueOnce(
+        Promise.resolve({
+          data: {
+            values: ['value1', 'value2'],
+          },
+        })
+      );
+    });
+
+    it('should return label names', async () => {
+      const query = 'label_names()';
+      const res = await ds.metricFindQuery(query);
+      expect(res.length).toBe(2);
+      expect(res[0].text).toEqual('value1');
+      expect(res[1].text).toEqual('value2');
+    });
+
+    it('should return label values', async () => {
+      const query = 'label_values(name)';
+      const res = await ds.metricFindQuery(query);
+      expect(res.length).toBe(2);
+      expect(res[0].text).toEqual('value1');
```

Comment:
Having the same values independent of the query looks a bit weird and does not tell you if you got the correct behaviour as it could actually call the wrong endpoint. We could mock `datasource.metadataRequest` and return different values depending on the url. There is actually already a mock for that in `/loki/mocks.ts`.